src/Microsoft.Diagnostics.Monitoring.WebApi/EgressValidationAttribute.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class)]
+    public sealed class EgressValidationAttribute : ActionFilterAttribute, IFilterFactory
+    {
+        public bool IsReusable => true;
+        IFilterMetadata IFilterFactory.CreateInstance(IServiceProvider serviceProvider)
+        {
+            var egressOutputOptions = serviceProvider.GetRequiredService<IEgressOutputOptions>();
+
+            EgressValidationActionFilter actionFilter = new EgressValidationActionFilter(egressOutputOptions);
+
+            return actionFilter;
+        }
+    }
+
+    public class EgressValidationUnhandledExceptionFilter : ActionFilterAttribute, IExceptionFilter
+    {
+        public void OnException(ExceptionContext context)
+        {
+            BadRequestObjectResult badRequestResult = new BadRequestObjectResult(context.Exception.Message);
```

Comment:
Not necessarily for this review, but we currently have a helper function that performs a similar translation (ActionContextExtensions). How do these two interact? Perhaps we can remove that helper entirely with an exception filter.

Reply:
> Perhaps we can remove that helper entirely with an exception filter.

Possibly, but we might lose logger scopes if we do that since it currently logs the exceptions in the `when` clause of the `catch` block in order to preserve the logger scopes. Something to keep in mind when IExceptionFilter is investigated.